python-avd/pyavd/_eos_designs/structured_config/overlay/router_bgp.py

```diff
@@ -24,457 +26,360 @@ class RouterBgpMixin(Protocol):
                 # TODO: (@Claus) told me to remove this
                 if self.shared_utils.evpn_role == "server":
-                    core_peer_group["default_route_target"] = {"only": True}
-                peer_groups.append(core_peer_group)
+                    peer_groups.obtain(self.inputs.bgp_peer_groups.evpn_overlay_core.name).default_route_target.only = True
 
             # Transposing the Jinja2 logic: if the evpn_overlay_core peer group is not
             # configured then the default_route_target is applied in the evpn_overlay_peers peer group.
             elif self.shared_utils.evpn_role == "server":
-                evpn_overlay_peers["default_route_target"] = {"only": True}
+                peer_groups.obtain(self.inputs.bgp_peer_groups.evpn_overlay_peers.name).default_route_target.only = True
 
         if self.shared_utils.overlay_routing_protocol == "ibgp":
             if self.shared_utils.overlay_mpls is True:
-                mpls_peer_group = {"name": self.inputs.bgp_peer_groups.mpls_overlay_peers.name, "activate": True}
+                peer_groups.append_new(name=self.inputs.bgp_peer_groups.mpls_overlay_peers.name, activate=True)
                 if self.shared_utils.evpn_role == "server" or self.shared_utils.mpls_overlay_role == "server":
-                    mpls_peer_group["default_route_target"] = {"only": True}
-                peer_groups.append(mpls_peer_group)
+                    peer_groups.obtain(self.inputs.bgp_peer_groups.mpls_overlay_peers.name).default_route_target.only = True
 
             if self.shared_utils.overlay_evpn_vxlan is True and (self.shared_utils.evpn_role == "server" or self.shared_utils.mpls_overlay_role == "server"):
-                evpn_overlay_peers["default_route_target"] = {"only": True}
-
-        peer_groups.append(evpn_overlay_peers)
-        address_family_rtc["peer_groups"] = peer_groups
+                peer_groups.obtain(self.inputs.bgp_peer_groups.evpn_overlay_peers.name).default_route_target.only = True
 
-        return address_family_rtc
-
-    def _address_family_vpn_ipvx(self: AvdStructuredConfigOverlayProtocol, version: int) -> dict | None:
+    def _set_address_family_vpn_ipvx(self: AvdStructuredConfigOverlayProtocol, version: int) -> None:
         if version not in [4, 6]:
             msg = "_address_family_vpn_ipvx should be called with version 4 or 6 only"
             raise AristaAvdError(msg)
 
         if (version == 4 and self.shared_utils.overlay_vpn_ipv4 is not True) or (version == 6 and self.shared_utils.overlay_vpn_ipv6 is not True):
-            return None
-
-        address_family_vpn_ipvx = {}
+            return
 
-        if self.shared_utils.overlay_ler is True or self.shared_utils.overlay_ipvpn_gateway is True:
-            address_family_vpn_ipvx["neighbor_default_encapsulation_mpls_next_hop_self"] = {"source_interface": "Loopback0"}
+        af_vpn = getattr(self.structured_config.router_bgp, f"address_family_vpn_ipv{version}")
```

Comment:
We loose all type safety with this. Could we just do a ternary option like `af_vpn = self.structured.config.router_bgp.address_family_vpn_ipv4 if version==4 else self.structured.config.router_bgp.address_family_vpn_ipv6`. That would give us a union type on the af_vpn and ensure that everything we do below will work with both.

Reply:
Fixed!